### PR TITLE
Remove `color_field` input type from profiles

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Clean up

## Description

Not that we have `Users::Setting` we no longer need color inputs for profiles. After talking to @benhalpern we decided to remove them.

## Related Tickets & Documents

https://github.com/forem/forem/pull/14206

## QA Instructions, Screenshots, Recordings

Nothing specific, color fields will get removed in the other PR.

### UI accessibility concerns?

n/a

## Added/updated tests?

- [X] Yes
 
## [Forem core team only] How will this change be communicated?

- [X] This change does not need to be communicated, and this is why not: this is pretty much an implementation detail at this point since customizable profiles have not been enabled for other Forems yet.

### CODE REVIEW

File: app/validators/profile_validator.rb [hunk 1]
```diff
@@ -42,10 +42,6 @@ def check_box_valid?(_record, _attribute)
     true # checkboxes are always valid
   end
 
-  def color_field_valid?(_record, _attribute)
```
Comment (citizen428) [inline app/validators/profile_validator.rb:45]: This dummy method was only recently added as an interim measure (deployed last Friday), it is now no longer needed.

File: app/views/users/_profile.html.erb [hunk 2]
```diff
@@ -107,20 +107,6 @@
             <label class="crayons-field__label" for="<%= "profile[#{field.attribute_name}]" %>">
               <%= field.label %>
             </label>
-          <% elsif field["input_type"] == "color_field" %>
```
Comment (citizen428) [inline app/views/users/_profile.html.erb:110]: We no longer need this whole special case block for handling color fields.

File: spec/fixtures/files/profile_fields.csv [hunk 3]
```diff
@@ -1,4 +1,4 @@
+Full test,text_field,Test,Test field,Basic,header,true
```
Comment (citizen428) [inline spec/fixtures/files/profile_fields.csv:1]: I update the test CSV data to remove color fields and also because I restructured the tests a little bit.

File: spec/services/profile_fields/import_from_csv_spec.rb [hunk 4]
```diff
@@ -10,31 +10,29 @@
   context "when missing attributes" do
     before { described_class.call(file_fixture("profile_fields.csv")) }
 
-    it "handles missing descriptions", :aggregate_failures do
-      field = ProfileField.find_by!(label: "Test name")
+    it "imports fields" do
```
Comment (citizen428) [inline spec/services/profile_fields/import_from_csv_spec.rb:13]: I restructured the tests here: the first one tests a full import of a well-defined field, the following specs test specific corner cases.

**Review by rhymes — APPROVED** [review]

**Review by Ridhwana — APPROVED** [review]
LGTM!
